migrate.py

```diff
@@ -390,6 +390,24 @@ def dir_to_path(path):
                 )
 
 
+    # multiple collections check
+    files_to_collections = defaultdict(list)
+    for ns, collections in spec.items():
+        for coll, plugin_types in collections.items():
+            for ptype, plugins in plugin_types.items():
+                for plugin in plugins:
+                    if ptype in ['modules', 'module_utils']:
+                        file_path = os.path.join('lib/ansible', ptype, plugin)
+                    else:
+                        file_path = os.path.join('lib/ansible/plugins', ptype, plugin)
+                    files_to_collections[file_path].append(coll)
+    dupes = {k: v for k, v in files_to_collections.items() if len(v) > 1}
+    if dupes:
+        err_msg = 'The following files are assigned to multiple collections:\n' + yaml.dump(dict(dupes))
```

Comment:
you created it via a dict comprehension, it's already a dict:
```suggestion
        err_msg = 'The following files are assigned to multiple collections:\n' + yaml.dump(dupes)
```

Reply:
Also, I think that "Each plugin needs to be assigned to one collection only." part is useful because it's an explanation of what needs to be done.